src/main/webapp/app/overview/discussion-section/discussion-section.component.ts

```diff
@@ -70,7 +70,11 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
             this.course = this.exercise?.course ?? this.lecture?.course;
             this.metisService.setCourse(this.course);
             this.metisService.setPageType(this.pageType);
-            this.setChannel(routeParams.params.courseId);
+            if (routeParams.params.courseId) {
```

Comment:
Why do we need to distinguish theses cases now?

Reply:
Since for the learning path view the courseId is not part of the URL but set by the parent container.